src/classes/BDI_DataImportService.cls

```diff
@@ -78,10 +78,14 @@ global with sharing class BDI_DataImportService {
     * @return DataImportBatch__c
     */
     private static DataImportBatch__c loadBatch(Id batchId) {
-
         List<DataImportBatch__c> dataImportBatches = BDI_DataImportBatch_SEL.selectByIds(new Set<Id>{ batchId });
         if (dataImportBatches.size() == 1) {
-            return dataImportBatches[0];
+            DataImportBatch__c currentBatch = dataImportBatches[0];
+            if (!String.isBlank(currentBatch.Active_Fields__c)) {
+                relevantDataImportFields = BGE_BatchGiftEntry_UTIL.getDataImportFields(batchId, false);
```

Comment:
This was the thing that made me the most uncomfortable about the original PR, but I am thinking we may be able to avoid it - applications calling BDI (like BGE) should ideally not be defining BDI behaviors, else we could build up all these conditionals and exceptions built in for each application that's calling into it. HOWEVER, I think we may be able to avoid this

Reply:
@ptewson-sfdo - Would moving this logic (along with other the field collection/map building) from the BGE util to the BDI_DataImportService class make sense? The one thing that immediately sticks out to me is that the `getActiveFieldNamesFromBatch` method from `BGE_BatchGiftEntry_UTIL` relies on the `BGE_ConfigurationWizard_CTRL.BGEField` class. If the above is acceptable, would moving `BGEField` into its own file make sense so we don't rely on the config wizard controller in BDI here as well?

Reply:
We believe we can pull the call to BGE_BatchGiftEntry_UTIL.getDataImportFields(batchId, false) entirely - these fields would only be used for FLS when the help text maps to a target object field and then the check would be on the target field. None of these fields have default mappings or map logically to target fields and would not be checked anyway, and we can remove an unneeded dependency on BGE by pulling this out.